packages/react-core/src/filters/tileFeaturesSpatialIndex.js

```diff
@@ -0,0 +1,94 @@
+import { indexes } from '@mapbox/tile-cover';
+// h3-js has a known problem that does not allow us to use it in a web-worker. To solve
+// it we're overwriting the node_module package after install it (check postinstall script in package.json)
+// To know more about the h3-js issue check https://github.com/uber/h3-js/issues/117
+import { h3GetResolution, polyfill } from 'h3-js';
+import bboxClip from '@turf/bbox-clip';
+import { SpatialIndex } from '../operations/constants/SpatialIndexTypes';
+
+export default function tileFeaturesSpatialIndex({
+  tiles,
+  geometryToIntersect,
+  spatialIndex
+}) {
+  const map = new Map();
+  const resolution = getResolution(tiles, spatialIndex);
+
+  if (!resolution) {
+    return [];
+  }
+  const cells = getCellsCoverGeometry(
+    geometryToIntersect.geometry,
+    spatialIndex,
+    resolution
+  );
+
+  if (!cells?.length) {
+    return [];
+  }
+  // We transform cells to object to improve the performace
+  const cellsDictionary = cells.reduce((prev, current) => {
+    prev[current] = {};
+    return prev;
+  }, {});
+
+  for (const tile of tiles) {
+    if (tile.isVisible === false || !tile.data) {
+      continue;
+    }
+
+    tile.data.forEach((d) => {
+      if (cellsDictionary[d.id]) {
+        map.set(d.id, d.properties);
+      }
+    });
+  }
+  return Array.from(map.values());
+}
+
+function getResolution(tiles, spatialIndex) {
+  const data = tiles.find((tile) => tile.data?.length)?.data;
+
+  if (!data) {
+    return;
+  }
+
+  if (spatialIndex === SpatialIndex.QUADINT || spatialIndex === SpatialIndex.QUADKEY) {
+    return data[0].id.length;
+  }
+
+  if (spatialIndex === SpatialIndex.H3) {
+    return h3GetResolution(data[0].id);
+  }
+}
+
+function getCellsCoverGeometry(geometry, spatialIndex, resolution) {
+  if (spatialIndex === SpatialIndex.QUADINT || spatialIndex === SpatialIndex.QUADKEY) {
+    return indexes(geometry, {
+      min_zoom: resolution,
+      max_zoom: resolution
+    });
+  }
+
+  if (spatialIndex === SpatialIndex.H3) {
+    const bboxWest = [-180, -90, 0, 90];
+    const bboxEast = [0, -90, 180, 90];
+
+    // The current H3 polyfill algorithm can't deal with polygon segments of greater than 180 degrees longitude
+    // so we clip the geometry to be sure that none of them is greater than 180 degrees
+    // https://github.com/uber/h3-js/issues/24#issuecomment-431893796
+    return polyfill(
```

Comment:
Will the resolution always be high relative to the viewport? I had to oversample in deck to avoid edge effects

Reply:
I don't understand very well, probably due to my lack of knowledge

Reply:
Perhaps I've got myself confused, I was just saying that doing it this way means we only capture hexagons that have their center in the polygon. If that is by design that is fine. 

For background: in deck, when filling the viewport so that I capture hexagons which are only partially in the viewport I polyfill at a higher resolution and then get the parents